lexicon/conversion.py

```diff
@@ -302,3 +303,109 @@ def run(self):
+
+        self.bliss_lexicon = bliss_lexicon
+        self.orth_mapping_file = orth_mapping_file
+        self.flip_mapping = flip_mapping
+        self.mapping_delimiter = mapping_delimiter
+
+        self.out_bliss_lexicon = self.output_path("lexicon.xml.gz")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        # load mapping from json or plain text file
+        is_json = self.orth_mapping_file.endswith(".json")
+        is_json |= self.orth_mapping_file.endswith(".json.gz")
+        if is_json:
+            with uopen(self.orth_mapping_file, "rt") as f:
+                mapping = json.load(f)
+            if self.flip_mapping:
+                mapping = {v: k for k, v in mapping.items()}
+        else:
+            mapping = dict()
+            with uopen(self.orth_mapping_file, "rt") as f:
+                for line in f:
+                    line = line.strip()
+                    if not line or line.startswith("#"):
+                        continue
+                    source_orth, target_orth = line.split(self.mapping_delimiter)
+                    if self.flip_mapping:
+                        source_orth, target_orth = target_orth, source_orth
+                    mapping[source_orth] = target_orth
+        print("\nA total of {} word mapping paris\n".format(len(mapping)))
+
+        # load input lexicon and build "orth to lemma" dict
+        lex = lexicon.Lexicon()
+        lex.load(self.bliss_lexicon.get())
+        orth2lemma = {}
+        for lemma in lex.lemmata:
+            primary_orth = lemma.orth[0]
+            if primary_orth in orth2lemma:
+                raise ValueError(
+                    "There shouldn't be two lemmata with the same primary "
+                    "orth, apply LexiconUniqueOrthJob before doing spelling "
+                    "conversion"
+                )
+            orth2lemma[primary_orth] = lemma
+
+        # spelling conversion
+        for source_orth, target_orth in mapping.items():
+            target_lemma = orth2lemma.get(target_orth, None)
+            source_lemma = orth2lemma.get(source_orth, None)
+            if target_lemma:
+                if source_lemma:
+                    for orth in source_lemma.orth:
+                        if orth not in target_lemma.orth:
+                            target_lemma.orth.append(orth)
+                    for phon in source_lemma.phon:
+                        if phon not in target_lemma.phon:
+                            target_lemma.phon.append(phon)
+                    for eval in source_lemma.eval:
+                        if eval not in target_lemma.eval:
+                            target_lemma.eval.append(eval)
+                    if not target_lemma.synt:
+                        if source_lemma.synt:
+                            target_lemma.synt = source_lemma.synt
+                        else:
+                            target_lemma.synt = source_orth.split()
```

Comment:
I am wondering if we should overwrite the `target_lemma.synt` even if it is set..?

Reply:
Makes sense for me to overwrite it anyway.
Words that have synt are mostly those which do not look like natural words, e.g. 'coz etc. 
Setting the synt anyway to American spelling cannot harm that much as we know the American spelling is definitely a valid synt we desire for this Job's original purpose. What do you think? @michelwi

Reply:
I think it makes sense to overwrite it. I don't think it will happen in practice.

Reply:
👍

Reply:
One question here: If I understand correctly you want to keep the original orthography for the language model (as presumably the LM is not updated when switching the spelling). Why would you override an already existing synt token with the original orthography? If the lemma already has a synt token isn't that what is expected to be in the LM and changing it to the split of the orthography is a bad idea?

Reply:
> Words that have synt are mostly those which do not look like natural words, e.g. 'coz etc.

In this Job we are dealing with natural words, so we think overwriting does no harm. (We need more evidence/observation though)

If @curufinwe @michelwi you both think we should revert to earlier decision, I can revert them to what it was like.

Reply:
I don't think this case will happen in practice and I don't have a strong opinion on which synt to keep.
We could also assert that only one synt is present and then the user has to decide on a per case basis.

Reply:
OK, then let's keep the original synt if it was defined already. Thanks!

Reply:
@curufinwe Done.